### PR TITLE
Auth deletion bug

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# Motivation and Context
db session handler wraps the response which seems to create unhandled exception

# What has changed
forced user delete to return 500 and 404 status code.

# How to test?

# Links

### CODE REVIEW

File: ras_party/controllers/respondent_controller.py [hunk 1]
```diff
@@ -81,12 +83,15 @@ def update_respondent_mark_for_deletion(email, session):
...
-        raise NotFound("Respondent does not exist")
-    logger.info("Marking respondent for deletion", email=obfuscate_email(email))
-    session.query(Respondent).filter(Respondent.party_uuid == respondent.party_uuid) \
-        .update({Respondent.mark_for_deletion: True})
+    if respondent:
+        try:
+            session.query(Respondent).filter(Respondent.party_uuid == respondent.party_uuid) \
+                .update({Respondent.mark_for_deletion: True})
+            return 'respondent successfully marked for deletion', 202
+        except (SQLAlchemyError, Exception):
```
Comment (warrenbailey) [inline ras_party/controllers/respondent_controller.py]: It's probably worth logging the error here so we can investigate

File: test/test_respondent_controller.py [hunk 2]
```diff
@@ -1494,11 +1494,11 @@ def test_put_change_respondent_account_status_bad_oauth2_response(self):
 
     def test_change_respondent_password_bad_auth_response(self):
         with patch('ras_party.controllers.account_controller.decode_email_token'), \
-             patch('ras_party.controllers.account_controller.current_app'), \
-             patch('ras_party.support.session_decorator.current_app.db'), \
-             patch('ras_party.controllers.account_controller.enrol_respondent_for_survey'), \
-             patch('ras_party.controllers.account_controller.NotifyGateway'), \
-             patch('ras_party.controllers.account_controller.OauthClient') as auth, \
+                patch('ras_party.controllers.account_controller.current_app'), \
```
Comment (warrenbailey) [inline test/test_respondent_controller.py:1497]: did you mean to change this indentation?

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: test/test_respondent_controller.py (hunk 2 above)
Comment (creativeappdevamit) [reply test/test_respondent_controller.py:1497]: no but the auto format for pycharm did and it looks good now

File: ras_party/controllers/respondent_controller.py (hunk 1 above)
Comment (creativeappdevamit) [reply ras_party/controllers/respondent_controller.py]: yes sure.. will update it.

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**